spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/ThreadPool.scala

```diff
@@ -41,6 +43,7 @@ class ThreadPool(private var poolSize: Int) {
 
   private def spawnThreadPool(poolSize: Int): ExecutionContext = {
     if (poolSize == 1) {
+      threadPool = MoreExecutors.sameThreadExecutor()
```

Comment:
looks like the threadPool is not used

Reply:
Looks like the sameThreadExecutor is deprecated?

Reply:
`invokeAndWait2` will use `threadPool` and the `default` pool size maybe 1.

Reply:
The `sameThreadExecutor` is deprecated for new version of guava and is replaced with `directExecutor`. But we use 11.0.2 which has only the `sameThreadExecutor`.

Reply:
I think the code structure is weird. And we should not use a deprecated method